### PR TITLE
feat: Added upgrade-python-requirements workflow

### PR DESCRIPTION
**Pull request by Jawayria**
JIRA: https://openedx.atlassian.net/browse/BOM-2600

### CODE REVIEW

**Review by mraarif — APPROVED** [review]

**Comment by Jawayria** [comment]
@edx/masters-devs-cosmonauts as this repository is owned by your team, you need to get access for this repo so that your team could get tagged on **Upgrade requirements prs** generated by this new GitHub workflow. 
Please review this PR as well.

**Comment by matthugs** [comment]
Added the @edx/masters-devs-cosmonauts team with Write permissions on this repo, such that the pings will work.

**Review by matthugs — CHANGES_REQUESTED** [review]
Could you apply the same feedback as I gave to https://github.com/edx/portal-designer/pull/135/files ? Although we never moved the jenkins-initiated upgrades over to alert us instead of Data Engineering, it'd be great to have this following the same pattern as all our other repos.

**Comment by Jawayria** [comment]
@matthugs @edx/edx-data-engineering please review

File: .github/workflows/upgrade-python-requirements.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+name: Upgrade Requirements
+
+on:
+  push:
+    branches:
+    - '**'
+  schedule:
+    # will start the job at 3:15 UTC on 15th and 30th of every month
+    - cron: "15 3 15,30 * *"
```
Comment (matthugs) [inline .github/workflows/upgrade-python-requirements.yml]:
```suggestion
    - cron: "15 3 15,29 * *"
```
```suggestion
    # will start the job at 3:15 UTC on 15th and 29th of every month
    - cron: "15 3 15,29 * *"
```

**Review by matthugs — APPROVED** [review]

**Pull request merged**